Sources/DBUS/DBusUnixFDClient.swift

```diff
@@ -0,0 +1,632 @@
+            let continuation = continuations.removeValue(forKey: replyTo)
+          {
+            continuation.yield(message)
+            continuation.finish()
+          } else if let replyTo = message.replyTo {
+            logger.warning(
+              "Received message with unknown reply-to",
+              metadata: [
+                "reply-to": "\(String(describing: replyTo))"
+              ])
+          } else if let handler = messageHandler {
+            await handler(message)
+          }
+        }
+      }
+
+      public func send(_ request: DBusRequest) async throws -> DBusMessage? {
+        try await send(request, timeoutNanoseconds: nil)
+      }
+
+      public func send(_ request: DBusRequest, timeoutNanoseconds: UInt64?) async throws
+        -> DBusMessage?
+      {
+        let requestId = await send.reserveSerial()
+        let message = DBusMessage(
+          byteOrder: request.byteOrder,
+          messageType: request.messageType,
+          flags: request.flags,
+          protocolVersion: request.protocolVersion,
+          serial: requestId,
+          headerFields: request.headerFields,
+          body: request.body,
+          unixFds: request.unixFds
+        )
+
+        if request.flags.contains(.noReplyExpected) {
+          logger.trace(
+            "Send request that does not expect a reply",
+            metadata: [
+              "serial": "\(requestId)"
+            ])
+          try await send.send(message)
+          return nil
+        }
+
+        logger.trace(
+          "Send request that expects a reply",
+          metadata: [
+            "serial": "\(requestId)"
+          ])
+        var replyContinuation: AsyncThrowingStream<DBusMessage, Error>.Continuation!
+        let replyStream = AsyncThrowingStream<DBusMessage, Error> { continuation in
+          replyContinuation = continuation
+        }
+        continuations[requestId] = replyContinuation
+        do {
+          try await send.send(message)
+        } catch {
+          continuations.removeValue(forKey: requestId)?.finish(throwing: error)
+          throw error
+        }
+
+        let iterator = replyStream.makeAsyncIterator()
+        do {
+          let reply = try await Self.awaitReply(
+            timeoutNanoseconds: timeoutNanoseconds,
+            iterator: ReplyIteratorBox(iterator: iterator)
+          )
+          return reply
+        } catch {
+          continuations.removeValue(forKey: requestId)?.finish(throwing: error)
+          throw error
+        }
+      }
+
+      public func setMessageHandler(
+        _ handler: @escaping @Sendable (DBusMessage) async -> Void
+      ) async {
+        messageHandler = handler
+      }
+
+      private struct ReplyIteratorBox: @unchecked Sendable {
+        let iterator: AsyncThrowingStream<DBusMessage, Error>.AsyncIterator
+      }
+
+      private static func awaitReply(
+        timeoutNanoseconds: UInt64?,
+        iterator: ReplyIteratorBox
```

Comment:
```suggestion
        iterator: inout ReplyIteratorBox
```

Maybe we can make it an inout. Some AsyncIterators expect their contents to never be copied

Reply:
That would also allow removing the copy at lines 202 and 210